src/ui/annotations.ts

```diff
@@ -1073,43 +1079,348 @@ export class AnnotationSchemaView extends Tab {
         this.updateProperty(oldProperty, newProperty);
       },
     });
-    this.element.appendChild(updateButton);
+    this.schemaActionButtons.appendChild(updateButton);
 
     const downloadButton = makeIcon({
-      text: "Download schema",
       title: "Download schema",
+      svg: svg_share,
       onClick: () => this.downloadSchema(),
     });
-    this.element.appendChild(downloadButton);
+    this.schemaActionButtons.appendChild(downloadButton);
 
     const copyButton = makeCopyButton({
       title: "Copy schema to clipboard",
       onClick: () => this.copySchemaToClipboard(),
     });
-    this.element.appendChild(copyButton);
+    this.schemaActionButtons.appendChild(copyButton);
 
     const pasteButton = makeIcon({
-      text: "Paste schema from clipboard",
       title: "Paste schema from clipboard",
+      svg: svg_clipboard,
       onClick: () => this.pasteSchemaFromClipboard(),
     });
-    this.element.appendChild(pasteButton);
+    this.schemaActionButtons.appendChild(pasteButton);
+  }
+
+  private createSchemaTable() {
+    const TABLE_HEADERS = ["Name", "Type", "Default value", ""];
+
+    const SCHEMA_TABLE_DATA = [
+      { name: "Name", type: "Enum (uint8)", defaultValue: "", deleteCell: "" },
+      { name: "Color alpha", type: "RGBa", defaultValue: "", deleteCell: "" },
+      { name: "is_deleted", type: "Boolean", defaultValue: "", deleteCell: "" },
+      { name: "Decimal", type: "float32", defaultValue: "", deleteCell: "" },
+      { name: "Color", type: "RGB", defaultValue: "", deleteCell: "" },
+      { name: "s0_start_x", type: "int8", defaultValue: "", deleteCell: "" }
+    ];
+
+    const DROPDOWN_OPTIONS = [
+      { header: "General", items: ["float32", "Boolean"] },
+      { header: "Enum", items: ["uint8", "uint16"] },
+      { header: "Colour", items: ["RGB", "RGBa"] },
+      { header: "Integer", items: ["int8", "int16", "int32"] }
+    ];
+
+    const SECTION_ICONS: Record<string, string> = {
+      "Enum": svg_format_size,
+      "Colour": svg_palette,
+      "Integer": svg_numbers
+    };
+
+    const ITEM_ICONS: Record<string, string> = {
```

Comment:
We could consider replacing the first string here by a literal for more informed type checking. So like `const ITEM_ICONS: Record<"float32" | "Boolean", string> = {"float32": svg_numbers, "Boolean": svg_check}`

Reply:
This could be similarly done for the SECTION_ICONS